src/NuGet.Core/NuGet.DependencyResolver.Core/GraphModel/GraphNode.cs

```diff
@@ -55,6 +55,19 @@ internal bool AreAllParentsRejected()
             return true;
         }
 
+        /// <summary>
+        /// Ensures that <see cref="InnerNodes"/> has the capacity to add
+        /// <paramref name="additionalSpace"/> more items.
+        /// </summary>
+        /// <param name="additionalSpace">The count of additional items that will be added.</param>
+        internal void EnsureInnerNodeCapacity(int additionalSpace)
+        {
+            if (InnerNodes is List<GraphNode<TItem>> innerList && innerList.Capacity < innerList.Count + additionalSpace)
+            {
+                innerList.Capacity = innerList.Count + additionalSpace;
```

Comment:
`innerList.Count + additionalSpace` is calculated twice. You could assign it to a variable and use it twice.